tst/unit/test_index_split.cpp

```diff
@@ -0,0 +1,157 @@
+#include "interface/meshblock_data.hpp"
+#include "interface/metadata.hpp"
+#include "interface/sparse_pack.hpp"
+#include "kokkos_abstraction.hpp"
+#include "mesh/meshblock.hpp"
+#include "parthenon/package.hpp"
+#include "utils/index_split.hpp"
+
+// TODO(jcd): can't call the MeshBlock constructor without mesh_refinement.hpp???
+#include "mesh/mesh_refinement.hpp"
+
+using namespace parthenon::package::prelude;
+using parthenon::BlockList_t;
+using parthenon::DevExecSpace;
+using parthenon::IndexDomain;
+using parthenon::IndexSplit;
+using parthenon::MeshBlock;
+using parthenon::MeshBlockData;
+using parthenon::MeshData;
+using parthenon::Metadata;
+using parthenon::PackIndexMap;
+using parthenon::par_for;
+using parthenon::Real;
+using parthenon::StateDescriptor;
+
+namespace {
+BlockList_t MakeBlockList(const std::shared_ptr<StateDescriptor> pkg, const int NBLOCKS,
+                          const int NSIDE, const int NDIM) {
+  BlockList_t block_list;
+  block_list.reserve(NBLOCKS);
+  for (int i = 0; i < NBLOCKS; ++i) {
+    auto pmb = std::make_shared<MeshBlock>(NSIDE, NDIM);
+    auto &pmbd = pmb->meshblock_data.Get();
+    pmbd->Initialize(pkg, pmb);
+    block_list.push_back(pmb);
+  }
+  return block_list;
+}
+// JMM: Variables aren't really needed for this test but...
+struct v1 : public parthenon::variable_names::base_t<false> {
+  template <class... Ts>
+  KOKKOS_INLINE_FUNCTION v1(Ts &&...args)
+      : parthenon::variable_names::base_t<false>(std::forward<Ts>(args)...) {}
+  static std::string name() { return "v1"; }
+};
+struct v3 : public parthenon::variable_names::base_t<false, 3> {
+  template <class... Ts>
+  KOKKOS_INLINE_FUNCTION v3(Ts &&...args)
+      : parthenon::variable_names::base_t<false, 3>(std::forward<Ts>(args)...) {}
+  static std::string name() { return "v3"; }
+};
+struct v5 : public parthenon::variable_names::base_t<false> {
+  template <class... Ts>
+  KOKKOS_INLINE_FUNCTION v5(Ts &&...args)
+      : parthenon::variable_names::base_t<false>(std::forward<Ts>(args)...) {}
+  static std::string name() { return "v5"; }
+};
+} // namespace
+
+TEST_CASE("IndexSplit", "[IndexSplit]") {
+  GIVEN("A set of meshblocks and meshblock and mesh data") {
+    constexpr int N = 6;
+    constexpr int NDIM = 3;
+    constexpr int NBLOCKS = 9;
+    constexpr int NG = 0;
+    const std::vector<int> scalar_shape{N, N, N};
+    const std::vector<int> vector_shape{N, N, N, 3};
+
+    Metadata m({Metadata::Independent, Metadata::WithFluxes}, scalar_shape);
+    Metadata m_vector({Metadata::Independent, Metadata::WithFluxes, Metadata::Vector},
+                      vector_shape);
+    auto pkg = std::make_shared<StateDescriptor>("Test package");
+    pkg->AddField(v1::name(), m);
+    pkg->AddField(v3::name(), m_vector);
+    pkg->AddField(v5::name(), m);
+    BlockList_t block_list = MakeBlockList(pkg, NBLOCKS, N, NDIM);
+
+    MeshData<Real> mesh_data("base");
+    mesh_data.Set(block_list, nullptr, NDIM);
+
```

Comment:
There's no test for `1<nkp<N` or `1<njp<N`? For when `nkp` or `njp` don't evenly divide into `N`?

Reply:
Added.